class.jetpack.php

```diff
@@ -3253,6 +3251,50 @@ public static function unlink_user( $user_id = null ) {
 		return true;
 	}
 
+	public static function update_disconnection_notice( $user_id ) {
+		$notice = Jetpack_Options::get_option( 'disconnection_notice' );
+		$notice = ! empty( $notice ) && is_array( $notice ) ? array_merge( $notice, array( $user_id ) ) : array( $user_id );
```

Comment:
Does this behave as expected? It seems to push duplicate user ids:

```php
var_export( array_merge( array_merge( [], [ 1 ], [ 1 ] ), [ 1 ] ) );
// array (
//  0 => 1,
//  1 => 1,
//  2 => 1,
// )
```

`array_unique` could solve that, could this could further simplified?

```php
	public static function update_disconnection_notice( $user_id ) {
		$notice = Jetpack_Options::get_option( 'disconnection_notice', array() );
		if ( ! in_array( $user_id, $notice ) ) {
			array_push( $notice, $user_id );
		}
		Jetpack_Options::update_option( 'disconnection_notice', $notice );
	}
```

Just sketching here, would it be possible that `$notice` isn't an array?